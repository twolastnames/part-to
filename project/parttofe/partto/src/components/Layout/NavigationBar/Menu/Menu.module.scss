@import "../../../../App.scss";

.navLink:hover {
  background-color: var(--background-color);
}
