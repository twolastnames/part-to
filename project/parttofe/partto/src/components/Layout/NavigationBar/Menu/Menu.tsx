import React from "react";

import classes from "./Menu.module.scss";
import { NavLink } from "@mantine/core";
import { Book, ChefHat, Icon, Info, Oven, Settings } from "../../../Icon/Icon";
import { ThemeSwitch } from "../ThemeSwitch/ThemeSwitch";

export function Menu() {
  return (
    <>
      <nav data-testid="Menu" className={classes.nav}>
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>Enter Recipe</div>}
          leftSection={<Icon definition={ChefHat} />}
        />
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>Start Meal</div>}
          leftSection={<Icon definition={Book} />}
        />
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>Continue Cooking</div>}
          leftSection={<Icon definition={Oven} />}
        />
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>Settings</div>}
          leftSection={<Icon definition={Settings} />}
        />
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>About</div>}
          leftSection={<Icon definition={Info} />}
        />
      </nav>
      <ThemeSwitch key={Math.random()} />
    </>
  );
}
