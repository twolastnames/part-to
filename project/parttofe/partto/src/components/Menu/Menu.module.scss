@import "../../App.scss";

.items {
  animation: fadein var(--animation-tempo);
}
