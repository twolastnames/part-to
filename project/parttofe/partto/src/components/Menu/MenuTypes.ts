export interface MenuProps {
  runState: string;
}
