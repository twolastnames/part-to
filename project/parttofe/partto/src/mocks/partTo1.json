{
  "workDuration": "1182.0",
  "clockDuration": "882.0",
  "name": "Bavarian Pot Roast",
  "tasks": [
    "cHsBXRyHSyBSSoCp8uXqr5",
    "jAykl0ltE7dytKfArBz2C3",
    "aekFmNJKUyt64YhBJwb2Z3",
    "qKJ0it3p4mgBc6mTIN58c5",
    "KI3qrAc3zpdd11F8m0AO93",
    "sPCQRKbqQBqxxFhNNdk0z6",
    "H3tjccSUkEfbmImH77jSk",
    "c972BQ4anGcTbMeMpxPv32"
  ]
}
