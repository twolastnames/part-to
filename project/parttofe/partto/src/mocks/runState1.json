{
  "runState": "wfGhVbmK0kzGYWqRZM2Xl3",
  "staged": [
    "jAykl0ltE7dytKfArBz2C3",
    "qKJ0it3p4mgBc6mTIN58c5",
    "KI3qrAc3zpdd11F8m0AO93",
    "sPCQRKbqQBqxxFhNNdk0z6",
    "c972BQ4anGcTbMeMpxPv32"
  ],
  "startTimes": [],
  "tasks": [],
  "duties": [],
  "started": [],
  "completed": [],
  "voided": [
    "cHsBXRyHSyBSSoCp8uXqr5",
    "aekFmNJKUyt64YhBJwb2Z3",
    "H3tjccSUkEfbmImH77jSk"
  ],
  "created": [],
  "activePartTos": ["PqgVxmgdJDZUJe7R13Yht"]
}
