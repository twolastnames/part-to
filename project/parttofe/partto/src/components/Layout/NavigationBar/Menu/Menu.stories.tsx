import React from "react";
import type { Meta, StoryObj } from "@storybook/react";

import { Menu } from "./Menu";
import { ShellProvider } from "../../../../ShellProvider";

const meta: Meta<typeof Menu> = {
  component: Menu,
  decorators: (Story) => (
    <ShellProvider>
      <Story />
    </ShellProvider>
  ),
};
export default meta;

type Story = StoryObj<typeof Menu>;

export const Simple: Story = {
  args: {},
};
