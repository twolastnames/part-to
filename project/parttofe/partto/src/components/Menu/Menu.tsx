import React from "react";

import classes from "./Menu.module.scss";
import { MenuProps } from "./MenuTypes";
import { useRunGet } from "../../api/runget";
import { Spinner } from "../Spinner/Spinner";
import { Accordion } from "../Accordion/Accordion";
import { useParttoGet } from "../../api/parttoget";

const Row = ({ partTo }: { partTo: string }) => {
  const response = useParttoGet({ partTo });
  return (
    <Spinner responses={[response]}>
      <li>{response.data?.name}</li>
    </Spinner>
  );
};

export function Menu({ runState }: MenuProps) {
  const response = useRunGet({ runState });

  return (
    <Spinner responses={[response]}>
      <Accordion summary="Menu">
        <ul data-testid="Menu" className={classes.items}>
          {response.data?.activePartTos?.map((partTo) => (
            <Row partTo={partTo} />
          ))}
        </ul>
      </Accordion>
    </Spinner>
  );
}
