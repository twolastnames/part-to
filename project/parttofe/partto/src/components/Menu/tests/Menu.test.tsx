import React from "react";
import { expect, test } from "@jest/globals";
import { render, screen, waitFor } from "@testing-library/react";
import { ShellProvider } from "../../../providers/ShellProvider";
import { Menu } from "../Menu";
import partTo1 from "../../../mocks/partTo1.json";
import runState1 from "../../../mocks/runState1.json";

test("snapshot", async () => {
  fetchMock.mockResponse((request: Request) => {
    if (request.url.includes("/api/parttos/")) {
      return Promise.resolve(
        JSON.stringify({
          partTos: ["partTo1"],
        }),
      );
    }
    if (request.url.includes("/api/partto/")) {
      return Promise.resolve(JSON.stringify(partTo1));
    }
    if (request.url.includes("/api/run/")) {
      return Promise.resolve(JSON.stringify(runState1));
    }
    return Promise.reject("");
  });
  render(
    <ShellProvider>
      <Menu runState="someRunStateId" />
    </ShellProvider>,
  );
  await waitFor(() => screen.findByTestId("Menu"));
  const component = screen.getByTestId("Menu");
  expect(component).toMatchSnapshot();
});
