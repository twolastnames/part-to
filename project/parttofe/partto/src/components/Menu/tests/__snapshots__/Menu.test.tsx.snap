// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<ul
  class="items"
  data-testid="Menu"
>
  <div
    class="error"
    data-testid="Spinner"
  />
</ul>
`;
